state/transaction.go

```diff
@@ -528,6 +528,12 @@ func (s *State) ParseTheTraceUsingTheTracer(evm *fakevm.FakeEVM, trace instrumen
 		// when a revert is detected, we stop the execution
 		if step.OpCode == "REVERT" {
 			stepError = fakevm.ErrExecutionReverted
+			// exit all contexts
+			if step.Depth > 1 {
+				for i := step.Depth; i > 1; i-- {
+					tracer.CaptureExit(step.ReturnData, gasCost.Uint64(), stepError)
+				}
```

Comment:
I'm not sure about this one, why we need to exit all the contexts???
May happen that a context revert, but the previous one continues the execution correctly!

Reply:
Yes, I've created 2 new tests to validate it and fixed it.